diff --git a/src/authorization.rs b/src/authorization.rs
index 9f71708..b79a1cd 100644
--- a/src/authorization.rs
+++ b/src/authorization.rs
@@ -15,7 +15,7 @@ pub struct AuthorizationSigningKey(SigningKey);
 #[derive(Clone, Copy, Debug, Default, Serialize, Deserialize, PartialEq)]
 pub struct AuthorizationVerifyingKey(AffinePoint);

-#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq)]
 pub struct AuthorizationSignature(Signature);

 impl AuthorizationSigningKey {
diff --git a/src/nullifier_key.rs b/src/nullifier_key.rs
index 9b345ae..e9fd795 100644
--- a/src/nullifier_key.rs
+++ b/src/nullifier_key.rs
@@ -4,7 +4,7 @@ use risc0_zkvm::sha::{Digest, Impl, Sha256, DIGEST_BYTES};
 use serde::{Deserialize, Serialize};

 /// Nullifier key
-#[derive(Clone, Serialize, Deserialize)]
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
 pub struct NullifierKey([u8; DIGEST_BYTES]);

 impl NullifierKey {
diff --git a/src/resource.rs b/src/resource.rs
index 5338f26..ccdbed3 100644
--- a/src/resource.rs
+++ b/src/resource.rs
@@ -29,7 +29,7 @@ use risc0_zkvm::Digest;
 use serde::{Deserialize, Serialize};

 /// A resource that can be created and consumed
-#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
 pub struct Resource {
     // a succinct representation of the predicate associated with the resource
     pub logic_ref: Digest,
